src/coreclr/src/gc/gc.cpp

```diff
@@ -7061,19 +7084,26 @@ void gc_heap::card_bundles_set (size_t start_cardb, size_t end_cardb)
     if (start_word < end_word)
     {
         // Set the partial words
-        card_bundle_table [start_word] |= highbits (~0u, card_bundle_bit (start_cardb));
+        uint32_t bits = highbits (~0u, card_bundle_bit (start_cardb));
+        set_bundle_bits (&card_bundle_table [start_word], bits);
 
         if (card_bundle_bit (end_cardb))
-            card_bundle_table [end_word] |= lowbits (~0u, card_bundle_bit (end_cardb));
+        {
+            bits = lowbits (~0u, card_bundle_bit (end_cardb));
+            set_bundle_bits (&card_bundle_table [end_word], bits);
+        }
 
         // Set the full words
         for (size_t i = start_word + 1; i < end_word; i++)
+        {
```

Comment:
Is this more efficient than a `memset()`?

Reply:
This codepath is used when compacting (copying) a contiguous plug of objects, which must be paired with copying the corresponding card table region and corresponding card bundle words. 

Each card bundle word here represents 8Mb of contiguous live objects.
Object plugs are generally not large enough to have a lot of iterations here, if at all. Thus `for` is preferable.